### ISSUE TITLE
AdaGCN forward model

### ISSUE THREAD

**Issue by Keiv4n** [pygda-team/pygda#6]
Hi, thanks for the great work on this project!

I think there is a potential bug in the forward_model() function for AdaGCN. Specifically, the loss computed during the adversarial training loop seems to be overwritten later in the method rather than being used.

Here’s the relevant code snippet:

```
for _ in range(10):
    encoded_source = self.adagcn(source_data)
    encoded_target = self.adagcn(target_data)

    gp_loss = self.gradient_penalty(encoded_source, encoded_target)

    dis_s = torch.mean(self.discriminator(encoded_source).reshape(-1))
    dis_t = torch.mean(self.discriminator(encoded_target).reshape(-1))
    dis_loss = - torch.abs(dis_s - dis_t)

    loss = dis_loss + self.gp_weight * gp_loss  # <-- this loss is overwritten below
```

Later in the function, this loss is redefined and returned:

```
cls_loss = self.adagcn.loss_func(source_logits, source_data.y)
dis_loss = torch.abs(dis_s - dis_t)

loss = cls_loss + dis_loss * self.domain_weight  # <-- overwrites the adversarial loss
```

To preserve the adversarial training loss, perhaps it should be stored or backpropagated separately (e.g., in a discriminator training step). Could you please clarify if this is the intended behavior, or should the adversarial loss be used in the final loss computation?

**Comment by cszhangzhen** [comment]
Hi,

Thanks for pointing this out.

There should be an adversarial loss in the loop. I will fix it as soon as possible.

Thanks!

**Comment by cszhangzhen** [comment]
The issue has been fixed in the main branch. Thank you.
